utils/serviceRequest.js

```diff
@@ -62,59 +63,118 @@ class ABServiceRequest extends ServiceCote {
-      const attempts = options.maxAttempts ?? ATTEMPT_REQUEST_MAXIMUM;
+      var attempts = options.maxAttempts ?? ATTEMPT_REQUEST_MAXIMUM;
 
-      let requestCount = 0;
+      let countRequest = 0;
 
       const paramStack = this.toParam(key, data);
       const domain = key.split(".")[0];
       const requester = this.getRequester(domain);
-      const sendRequest = async () => {
-         requestCount += 1;
-         try {
-            const results = await requester.send({
-               ...paramStack,
-               __timeout: timeout,
-            });
-            return results;
-         } catch (err) {
-            if (
-               err.message === "Request timed out." &&
-               requestCount < attempts
-            ) {
-               this.req.log(
-                  `... timeout waiting for request (${key}), retrying ${requestCount}/${attempts}`
-               );
-               return await sendRequest();
-            } else {
-               throw err;
-            }
-         }
-      };
-      try {
-         const results = await sendRequest();
-         if (this.req.performance) {
-            this.req.performance.measure(key, key);
-         }
-         if (callback) return callback(null, results);
-         return results;
-      } catch (err) {
-         err._serviceRequest = key;
-         err._params = paramStack;
-         if (key != "log_manager.notification") {
-            // If there was an error requesting log_manger.notification, don't use req.notify or we can get caught in an infinite loop
-            this.req.notify.developer(err, {
-               message: `Could not request (${key}) - ${JSON.stringify(
-                  paramStack
-               )}`,
-            });
-         }
-         if (callback) return callback(err);
-         throw err;
-      }
+
+      var timeoutCleanup = false;
+      // {bool}
+      // are we attempting to clean up a timedout service call?
+
+      let returnPromise = new Promise((resolve, reject) => {
+         const sendRequest = () => {
+            countRequest += 1;
+
+            requester.send(
+               {
+                  ...paramStack,
+                  __timeout: timeout,
+               },
+               (err, results) => {
+                  let finalTime;
+                  if (this.req.performance) {
+                     finalTime = this.req.performance.measure(key, key);
+                  }
+
+                  if (err) {
+                     // https://github.com/dashersw/cote/blob/master/src/components/requester.js#L132
+                     if (err.message === "Request timed out.") {
+                        // Retry .send
+                        if (!timeoutCleanup && countRequest < attempts) {
+                           this.req.log(
+                              `... timeout waiting for request (${key}), retrying ${countRequest}/${ATTEMPT_REQUEST_MAXIMUM}`
```

Comment:
I think it should be `${countRequest}/${attempts}`.
 `ATTEMPT_REQUEST_MAXIMUM` can be overridden by `options.maxAttempts`